rfcs/000-resource-notices.md

```diff
@@ -0,0 +1,88 @@
+
+## Guide-level Explanation (aka Product Details)
+
+An Amundsen final user can see a notice in the table/dashboard/user detail page whenever their system admin adds one related to it.
+
+A notice is a small box with an icon and a message containing HTML markup (like links and bolded text).
+
+These will come in three flavors:
+
+- Informational: Marked with a blue "i" icon on the right side
+- Warning: Marked with an orange exclamation mark icon on the right side
+- Alert: Marked with a red exclamation mark icon on the right side
+
+For example, if company X wants to deprecate the use of one table, they can opt to add a new notice in their configuration file:
+
+```
+    ... Table Configuration
+    notices: {
+        "<SCHEMANAME>.<TABLENAME>": {
+          severity: NoticeSeverity.ALERT,
+          message: `<span>This table is deprecated, please use <a href="<LINKTONEWTABLEDETAILPAGE>">this new table</a> instead.</span>`,
+        },
+    },
+```
+
+This code will show a notice with a red exclamation icon whenever a final user visits the table's Table Detail page.
+
+This feature's ultimate goal is to allow Amundsen administrators to point their users to more trusted/higher quality data without removing the old references.
+
+## Reference-level Explanation (aka Technical Details)
+
+The design we propose is based on the following:
+
+- Using our current frontend configuration files (config-utils.ts, config-default.ts, config-custom.ts, and config-types.ts)
+- Allowing a three-level "severity" schema for the alerts (info, warning, and alert)
+- Allowing its use in "tables" and "dashboards" (we could add it to people as well, but we'll need designs for it)
+
+What we are not supporting for now:
+
+- Multiple notifications per resource
+- Loading of external files (like a JSON)
+- Application of a single notice to multiple resources
+
+In the implementation, we'll check the notices object whenever we visit the resource detail page, showing any notification if present.
+
+It is not a breaking change, there won't be any costs for older versions to support this.
+
+## Drawbacks
+
+- This implementation could not be scalable, as maybe users want to have too many notifications. It doesn't mean this work won't be a good start, but it could be not enough. TBD
+- With this approach, to add/update/remove a notification, admins would need to make a new deployment of the application. A better approach would be if we use a UI, so a proper feature flag system will be much better for this.
+- Notices are on a 1:1 basis. If admins wanted to deprecate a whole schema or dashboard product, they would need to create a notice for each table/dashboard.
+
+## Alternatives
+
+- Integrating a proper feature flag system and feed the configuration from it. This would be ideal not just for this feature but for many others. It is also more complex and costly.
```

Comment:
Is feature flagging the right type of system? imo feature flags means turning on or off certain features for a population, for example for a/b testing. This seems more like administrator UI or administrator privileges

Reply:
True, maybe feature flags won't be the best match, but they would allows us to change them on the fly.

Yeah, an admin UI would be a better fit, but I don't know if we have more use cases for it, worth to keep it in mind!